compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3288,6 +3288,12 @@ private boolean equalityIntersectionExists(Set<BType> lhsTypes, Set<BType> rhsTy
      * @return  a set containing all the retrieved member types
      */
     public Set<BType> expandAndGetMemberTypesRecursive(BType bType) {
+        LinkedHashSet<BType> visited = new LinkedHashSet<>();
```

Comment:
Any specific reason to use a linked hashset here?

Reply:
Thank you for the suggestion.

Reply:
Fixed by b662fb58db3762fd4bfc3811605bcf6d70f53c7e